pkg/infra/tracing/test_helper.go

```diff
@@ -1,7 +1,9 @@
 package tracing
 
 func InitializeTracerForTest() (Tracer, error) {
-	ots := &Opentelemetry{}
+	ots := &Opentelemetry{
+		enabled: "jaeger",
```

Comment:
I think there's one tracer for testing, keeping traces in memory and allowing for later retrieval. If this is interesting to you, let me know and I'll try to find more details.